tests/test_nn/test_modules_literal.py

```diff
@@ -0,0 +1,47 @@
+# -*- coding: utf-8 -*-
+
+"""Tests for literal interaction functions."""
+
+import torch
+from torch import nn
+
+from pykeen.datasets.nations import NationsLiteral
+from pykeen.nn.modules import ComplExInteraction, LiteralInteraction
+from pykeen.triples.triples_numeric_literals_factory import TriplesNumericLiteralsFactory
+from tests import cases
+
+nations_literal = NationsLiteral()
+
+
+class MockLiteralInteraction(LiteralInteraction):
```

Comment:
@mberr couldnyou help look into why these tests aren't passing?

Reply:
@cthoyt now some tests do not pass since the `LiteralInteraction` is not a subclass of `FunctionalInteraction` and thus does not have `_prepare_for_functional`

<summary>Traceback</summary>
<details>

```python-traceback
tests/cases.py:457 (LiteralTests.test_forward_consistency_with_functional)
self = <tests.test_nn.test_modules_literal.LiteralTests testMethod=test_forward_consistency_with_functional>

    def test_forward_consistency_with_functional(self):
        """Test forward's consistency with functional."""
        # set in eval mode (otherwise there are non-deterministic factors like Dropout
        self.instance.eval()
        for hs, rs, ts in self._get_test_shapes():
            h, r, t = self._get_hrt(hs, rs, ts)
            scores = self.instance(h=h, r=r, t=t)
>           kwargs = self.instance._prepare_for_functional(h=h, r=r, t=t)

../cases.py:465: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = LiteralInteraction(
  (base): ComplExInteraction()
  (combination): Sequential(
    (0): Linear(in_features=4, out_features=2, bias=True)
    (1): Dropout(p=0.1, inplace=False)
  )
)
name = '_prepare_for_functional'

    def __getattr__(self, name: str) -> Union[Tensor, 'Module']:
        if '_parameters' in self.__dict__:
            _parameters = self.__dict__['_parameters']
            if name in _parameters:
                return _parameters[name]
        if '_buffers' in self.__dict__:
            _buffers = self.__dict__['_buffers']
            if name in _buffers:
                return _buffers[name]
        if '_modules' in self.__dict__:
            modules = self.__dict__['_modules']
            if name in modules:
                return modules[name]
>       raise AttributeError("'{}' object has no attribute '{}'".format(
            type(self).__name__, name))
E       AttributeError: 'LiteralInteraction' object has no attribute '_prepare_for_functional'

../../../../venv/pykeen/lib/python3.8/site-packages/torch/nn/modules/module.py:947: AttributeError
```
</details>

Reply:
is there a reason we need these? do we need to modify these tests, or can I just override them to not test anything?

Reply:
I suppose we do not need them. :sweat_smile:

Reply:
oky I think I solved it

Reply:
in the monotonic affine transform class, these were just clobbered, but I added some checks inside the parent class itself to automatically skip these tests if not possible